### DISCUSSION TITLE
ChunkLoadError: Loading chunk (...)

### DISCUSSION THREAD

**Discussion by GetWrecked99** [vercel/next.js#51277] [Help] [answered]
### Summary

i'm trying to implement protected routes using middleware and cookies (cookies-next lib).
currently using next v13.4.5 and appDir as well. 
the problem is: 
after logging in, when i clicked the logout button to firstly remove the cookie then navigate to the root page, everything works fine but after reloading the page getting this error :

Uncaught ChunkLoadError: Loading chunk app/(Authenticated)/dashboard/page failed.
(missing: http://localhost:3000/_next/static/chunks/app/(Authenticated)/dashboard/page.js)
    at __webpack_require__.f.j (webpack.js:868:29)
    at webpack.js:155:40
    at Array.reduce (<anonymous>)
    at __webpack_require__.e (webpack.js:154:67)
    at fn.e (webpack.js:406:50)
    at self.__next_chunk_load__ (webpack-internal:///(:3000/app-client)/./node_modules/next/dist/client/app-index.js:70:32)
    at preloadModule (webpack-internal:///(:3000/app-client)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:85:33)
    at resolveModule (webpack-internal:///(:3000/app-client)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:1572:17)
    at processFullRow (webpack-internal:///(:3000/app-client)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:1642:9)
    at processBinaryChunk (webpack-internal:///(:3000/app-client)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:1693:5)
    at progress (webpack-internal:///(:3000/app-client)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:1747:5)

### Additional information

```js
middleware setup : 

import { NextResponse, type NextRequest } from 'next/server'

export const middleware = (request: NextRequest) => {
  const { cookies } = request
  const { url } = request

  const token = cookies.get('token')

  if (!token && url.includes('dashboard')) {
    return NextResponse.redirect('http://localhost:3000/')
  }

  return NextResponse.next()
}


package.json :

{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@ant-design/colors": "^7.0.0",
    "@fortawesome/fontawesome-svg-core": "^6.4.0",
    "@fortawesome/free-regular-svg-icons": "^6.4.0",
    "@fortawesome/free-solid-svg-icons": "^6.4.0",
    "@fortawesome/react-fontawesome": "^0.2.0",
    "@hookform/error-message": "^2.0.1",
    "@hookform/resolvers": "^3.1.0",
    "@reduxjs/toolkit": "^1.9.5",
    "@svgr/webpack": "^8.0.1",
    "@types/node": "20.1.7",
    "@types/react": "^18.2.8",
    "@types/react-dom": "18.2.4",
    "@types/react-redux": "^7.1.25",
    "@typescript-eslint/eslint-plugin": "^5.59.6",
    "antd": "^5.5.0",
    "antd-button-color": "^1.0.4",
    "antd-jalali": "^1.0.3",
    "autoprefixer": "10.4.14",
    "cookies-next": "^2.1.2",
    "eslint": "8.40.0",
    "eslint-config-next": "13.4.2",
    "i18next": "^22.5.0",
    "i18next-browser-languagedetector": "^7.0.1",
    "next": "^13.4.5",
    "postcss": "8.4.23",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.44.3",
    "react-i18next": "^12.3.1",
    "react-redux": "^8.0.5",
    "redux": "^4.2.1",
    "redux-persist": "^6.0.0",
    "sass": "^1.62.1",
    "tailwind-scrollbar": "^3.0.4",
    "tailwindcss": "3.3.2",
    "typescript": "5.0.4",
    "yup": "^1.2.0"
  },
  "devDependencies": {
    "eslint-config-prettier": "^8.8.0",
    "prettier": "^2.8.8"
  },
  "resolutions": {
    "@types/react": "^18.2.8",
    "@types/react-dom": "18.2.4"
  }
}


and the next.config file : 

const path = require('path')

/** @type {import('next').NextConfig} */

const nextConfig = {
  reactStrictMode: true,
  sassOptions: {
    includePaths: [path.join(__dirname, 'src/styles')],
  },
  webpack(config) {
    // Grab the existing rule that handles SVG imports
    const fileLoaderRule = config.module.rules.find((rule) => rule.test?.test?.('.svg'))

    config.module.rules.push(
      // Reapply the existing rule, but only for svg imports ending in ?url
      {
        ...fileLoaderRule,
        test: /\.svg$/i,
        resourceQuery: /url/, // *.svg?url
      },
      // Convert all other *.svg imports to React components
      {
        test: /\.svg$/i,
        issuer: /\.[jt]sx?$/,
        resourceQuery: { not: /url/ }, // exclude if *.svg?url
        use: ['@svgr/webpack'],
      }
    )

    // Modify the file loader rule to ignore *.svg, since we have it handled now.
    fileLoaderRule.exclude = /\.svg$/i

    return config
  },
}

module.exports = nextConfig
```

**Comment by icyJoseph** [comment, marked as answer]
I think you probably want to opt out the `_next/` internals from the middleware. 


```js
export const config = {
  matcher: [
    /*
     * Match all request paths except for the ones starting with:
     * - api (API routes)
     * - _next/static (static files)
     * - _next/image (image optimization files)
     * - favicon.ico (favicon file)
     */
    '/((?!api|_next/static|_next/image|favicon.ico).*)',
  ],
}
```

Because, otherwise, you reject `/_next/static/chunks/app/(Authenticated)/dashboard/page.js` because it includes dashboard.

**Reply by GetWrecked99** [reply]
it works !! appreciated mate :) 
but honestly i tried too many solutions in the past 3 days. The weirdest thing was that i used the same implementation in a new project with the same version on next, cookie lib, also same routing style (with the exact routing group patterns) but didn't get that error. that was working exactly fine ..
